components/identities/actions/identityActions.tsx

```diff
@@ -148,12 +154,20 @@ const IdentityActions: FunctionComponent<IdentityActionsProps> = ({
 
   useEffect(() => {
     if (isAutoRenewalEnabled) {
-      setDisableRenewalCalldata(
-        autoRenewalCalls.disableRenewal(callDataEncodedDomain[1].toString())
-      );
+      const disableCallData: Call[] = [];
+      autoRenewalData.forEach((renewalData) => {
+        disableCallData.push(
+          autoRenewalCalls.disableRenewal(
+            renewalData.auto_renew_contract ??
+              (process.env.NEXT_PUBLIC_RENEWAL_CONTRACT as string),
```

Comment:
why `renewalData.auto_renew_contract` could be undefined? if AutoRenewalEnable is true we shouldn't even have this. 

Maybe be js doesn't see it so you can add `renewalData` in the `  if (isAutoRenewalEnabled)` cause it makes this code more difficult to understand and add the NEXT_PUBLIC_RENEWAL_CONTRACT